### PR TITLE
Gradle update

### LINKED ISSUE #164: build gradle upgrade
**Is your feature request related to a problem? Please describe.**
Here are two main reasons to upgrade our build file.  Firstly we try to solve the tool version problem, it also describes on #159. Secondly, our current build file was run on Gradle 2 to 4, but the newest version is 6.4 already, many features, methods removed soon on newer Gradle. 

**Describe the solution you'd like**
We will upgrade our current build Gradle to address below two issues:
1. Tool version issue was discussed in our group meeting.  All projects under AdamaJava will be assigned the same version which is the number from a counter pulsing first five-digit from revision.  The counter will auto-increment during each time merging into the master branch successfully. 

2. upgrade build file to Gradle 6.4 with below changes:
 a. replace deprecated variables and methods:  "compile" to "api"; "testCompile" to "testImplementation"; "configurations.compile" to "configurations.runtimeClasspath" ; 
 b. replace plugins: "findbugs" to "spotbugs"; "java" to "java-library"

**Describe alternatives you've considered**
additional tidy up is done: 
1. tidy up build file, remove unnecessary functions: stopping create <subproject>/adama folder. 

2. tidy up subproject build file:  
 a.  such as "compile" to "api",  testCompile" to "testImplementation";
 b. remove sourceSets.test.compileClasspath and runtimeClasspath, because it already done in root build.gradle; 
 c. remove fatjar task from qprofiler2, due to the default dist task do the similar thing;
 d. remove sourceSets from q3indel, due to the java srcDirs already set on root build.gradle, and there is no scala required in q3indel.

### PR DESCRIPTION
**Pull request by ChristinaXu2017**
# Description

Here we update our build file to Gradle 6.4. Hence, the new build will apply a universal version to root and sub-projects, which is the current counter from version.txt plus short git version hash. Meanwhile, we updated the unit test, the main reason is TemporaryFolder usage is quite different between junit-4.12 and junit-4.10. 

Fixes  #164 

## Type of change
Heavy changes on build files and unit tests under qSV, qpileup, qbasepileup.

- [X ] Bug fix (non-breaking change which fixes an issue):  Unit test under qSV, qpileup, qbasepileup are heavily changed due to the TemporaryFolder usage updating. The previous code won't work properly with the current junit-4.12. 
- [X ] New feature (non-breaking change which adds functionality): all build files are updated, major changes occur on root build file. Now build file utilize the new features from Gradle 6.4.
- [ X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X ] This change requires a documentation update

# How Has This Been Tested?
We build it locally and then run the individual tool,  eg. "java -jar qprofiler2.jar --version", make sure current universal version is assigned through the build manifest process, at the moment the version is "7-4e22d0b7". Unit tests are also updated and tested. 

# Checklist:

- [X ] My code follows the style guidelines of this project
- [X ] I have performed a self-review of my own code
- [X ] I have commented on my code, particularly in hard-to-understand areas
- [X ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings: the updated "spotbugs" may generate different warning messages.
- [ X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: qpileup/test/org/qcmg/pileup/OptionsTest.java [hunk 1]
```diff
@@ -1,8 +1,9 @@
-package org.qcmg.qpileup;
+package org.qcmg.pileup;
```
Comment (holmeso) [inline qpileup/test/org/qcmg/pileup/OptionsTest.java:1]: any reason the package name is changing?

File: qpileup/test/org/qcmg/pileup/MessagesTest.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package org.qcmg.qpileup;
+package org.qcmg.pileup;
```
Comment (holmeso) [inline qpileup/test/org/qcmg/pileup/MessagesTest.java:1]: any reason the package name is changing?

File: qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java [hunk 3]
```diff
@@ -46,14 +46,19 @@
...
     public static File iniFile;
     public static File logFile;
     public static File outputDir;
 
     @Before
     public void setUp() throws IOException {        
-      	normalBam = testFolder.newFile("normal.bam");    	
+      	normalBam = testFolder.newFile("normal.bam"); 
+      	iniFile = testFolder.newFile("test.ini");
+//		File logFile = testFolder.newFile("test.log");
```
Comment (holmeso) [inline qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java:59]: remove

File: qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java [hunk 4]
```diff
@@ -63,17 +68,18 @@ public void tearDown() {
...
+		File logFile = testFolder.newFile("test.log");
 		File controlBam =  testFolder.newFile("control.bam");
 		File testBam =  testFolder.newFile("test.bam");
 		outputDir = testFolder.newFolder();
 		File reference = testFolder.newFile("reference_file");
-		if (iniFile.exists()) {
-			iniFile.delete();
-		}		
+		
+//		File iniFile = testFolder.newFile("test.ini");
```
Comment (holmeso) [inline qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java:79]: remove

File: qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java [hunk 5]
```diff
@@ -113,13 +119,13 @@ private static String setUpIniFile() throws IOException {
...
 		}
-		return iniFile.getAbsolutePath();
+		 
 	}
     
-	public static String[] getValidOptions() throws IOException {
-			String iniFile = setUpIniFile();
-            return new String[] {"--ini", iniFile, "--tmp",  testFolder.getRoot().toString()};
-    }
+//	public static String[] getValidOptions() throws IOException {
```
Comment (holmeso) [inline qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java:125]: remove

File: qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java [hunk 6]
```diff
@@ -130,7 +136,8 @@ public void setupQueryExecutor() throws Exception {
 		CountDownLatch fLatch = null;
 		CountDownLatch wGoodLatch = null;
 		
-	    String[] args = getValidOptions();
+	   // String[] args = getValidOptions();
```
Comment (holmeso) [inline qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java:139]: remove

File: qsv/test/org/qcmg/qsv/util/TestUtil.java [hunk 7]
```diff
@@ -54,60 +54,58 @@
...
-	private static String setUpIniFile(final TemporaryFolder testFolder, String preprocessMode,
+	private static String setUpIniFile(final File testFolder, String preprocessMode,
 			final String analysisMode, String normalBam, String tumorBam, int clusterSize, int filterSize, String tmpDir, String outputDir, boolean goodOutput, String mapper, String sequencingPlatform) throws IOException {
 		
-		File iniFile = testFolder.newFile("test.ini");
-		File reference = testFolder.newFile("reference_file");
-		if (iniFile.exists()) {
-			iniFile.delete();
-		}		
+//		File iniFile = testFolder.newFile("test.ini");
```
Comment (holmeso) [inline qsv/test/org/qcmg/qsv/util/TestUtil.java:102]: remove

File: qsv/test/org/qcmg/qsv/util/TestUtil.java [hunk 8]
```diff
@@ -833,14 +840,18 @@ public static Clip getClip(String strand, String side, String name) {
...
 	      pairs.add(new MatePair("HWI-ST1240:47:D12NAACXX:4:1304:9295:38861:20120608092353631,chr10,89700053,89700153,AAC,97,false,HWI-ST1240:47:D12NAACXX:4:1304:9295:38861:20120608092353631,chr10,89712340,89712440,AAC,145,true,F1R2"));
 	      pairs.add(new MatePair("HWI-ST1240:47:D12NAACXX:7:1309:3779:55661:20120608113919562,chr10,89700060,89700160,AAC,97,false,HWI-ST1240:47:D12NAACXX:7:1309:3779:55661:20120608113919562,chr10,89712340,89712440,AAC,145,true,F1R2"));
 	      pairs.add(new MatePair("HWI-ST1240:47:D12NAACXX:6:2109:5161:48848:20120608110941621,chr10,89700064,89700164,AAC,97,false,HWI-ST1240:47:D12NAACXX:6:2109:5161:48848:20120608110941621,chr10,89712346,89712446,AAC,145,true,F1R2"));
 	      pairs.add(new MatePair("HWI-ST1240:47:D12NAACXX:8:1109:11792:69957:20120608020343585,chr10,89700200,89700300,AAC,161,false,HWI-ST1240:47:D12NAACXX:8:1109:11792:69957:20120608020343585,chr10,89712446,89712546,AAC,81,true,F2R1"));
-	      String tumourFile = testFolder.newFile("tumor.bam").getAbsolutePath();
+	     // String tumourFile = testFolder.newFile("tumor.bam").getAbsolutePath();
+	      File f = new File(testFolder,"tumor.bam" );
```
Comment (holmeso) [inline qsv/test/org/qcmg/qsv/util/TestUtil.java:851]: why is this file being created?

File: qbasepileup/test/org/qcmg/qbasepileup/indel/IndelPileupForDeletionTest.java [hunk 9]
```diff
@@ -154,7 +158,8 @@ public void testGetDeletionIndelLength() throws QBasePileupException, IOExceptio
 		expect(options.includeDuplicates()).andReturn(false);
 		replay(options);
 		position = new IndelPosition(line, false, "pindel", cols);
-		pileup = new IndelPileup(options, new InputBAM(testFolder.newFile("tumour.bam")), position, testFolder.newFile("reference.fa"), 5, 10, 3, true);
+//		pileup = new IndelPileup(options, new InputBAM(testFolder.newFile("tumour.bam")), position, testFolder.newFile("reference.fa"), 5, 10, 3, true);
```
Comment (holmeso) [inline qbasepileup/test/org/qcmg/qbasepileup/indel/IndelPileupForDeletionTest.java:161]: remove commented out code

File: qpileup/test/org/qcmg/pileup/OptionsTest.java [hunk 10]
```diff
@@ -25,37 +26,52 @@
...
+
 	}
 	
 	@Test
 	public void testValidViewOptions() throws Exception {
 		Options options = TestUtil.getValidOptions(testFolder, "view", reference, hdf, bam, testFolder.getRoot().toString(), "all", "");
 		options.parseIniFile();
 		assertEquals("view", options.getMode());
-		assertEquals(testFolder.getRoot().getAbsolutePath() + PileupConstants.FILE_SEPARATOR + "log", options.getLog());
+//		assertEquals(testFolder.getRoot().getAbsolutePath() + PileupConstants.FILE_SEPARATOR + "log", options.getLog());
```
Comment (holmeso) [inline qpileup/test/org/qcmg/pileup/OptionsTest.java:66]: remove

File: qpileup/test/org/qcmg/pileup/OptionsTest.java [hunk 11]
```diff
@@ -64,19 +80,24 @@ public void testValidMergeOptions() throws Exception {
 		Options options = TestUtil.getValidOptions(testFolder, "merge", reference, mergeHDF, bam, testFolder.getRoot().toString(), "all", hdf);
 		options.parseIniFile();
 		assertEquals("merge", options.getMode());
-		assertEquals(testFolder.getRoot().getAbsolutePath() + PileupConstants.FILE_SEPARATOR + "log", options.getLog());
+//		assertEquals(testFolder.getRoot().getAbsolutePath() + PileupConstants.FILE_SEPARATOR + "log", options.getLog());
```
Comment (holmeso) [inline qpileup/test/org/qcmg/pileup/OptionsTest.java:83]: remove

File: qpileup/test/org/qcmg/pileup/OptionsTest.java [hunk 12]
```diff
@@ -64,19 +80,24 @@ public void testValidMergeOptions() throws Exception {
...
+		
 	}
 	
 	@Test
 	public void testValidMetricsOptions() throws Exception {		
 		Options options = TestUtil.getValidOptions(testFolder, "metrics", reference, hdf, bam, testFolder.getRoot().toString(), "all", null);
 		options.parseIniFile();
 		assertEquals("metrics", options.getMode());
-		assertEquals(testFolder.getRoot().getAbsolutePath() + PileupConstants.FILE_SEPARATOR + "log", options.getLog());
+//		assertEquals(testFolder.getRoot().getAbsolutePath() + PileupConstants.FILE_SEPARATOR + "log", options.getLog());
```
Comment (holmeso) [inline qpileup/test/org/qcmg/pileup/OptionsTest.java:100]: remove

File: qpileup/test/org/qcmg/pileup/OptionsTest.java (hunk 1 above)
Comment (ChristinaXu2017) [reply qpileup/test/org/qcmg/pileup/OptionsTest.java:1]: I think there was typo, there are two folder under package org.qcmg for "qpileup",  but there is only one folder under other tools.  There is no reason have two different package names, they should be either "org.qcmg.qpileup" or "org.qcmg.pileup".

File: qpileup/test/org/qcmg/pileup/MessagesTest.java (hunk 2 above)
Comment (ChristinaXu2017) [reply qpileup/test/org/qcmg/pileup/MessagesTest.java:1]: as above

File: qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java (hunk 3 above)
Comment (ChristinaXu2017) [reply qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java:59]: done

File: qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java (hunk 4 above)
Comment (ChristinaXu2017) [reply qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java:79]: done

File: qpileup/test/org/qcmg/pileup/OptionsTest.java (hunk 1 above)
Comment (holmeso) [reply qpileup/test/org/qcmg/pileup/OptionsTest.java:1]: sounds good, so have all tests now been moved from qpileup -> pileup?

File: qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java (hunk 5 above)
Comment (ChristinaXu2017) [reply qsv/test/org/qcmg/qsv/annotate/AnnotateFilterMTTest.java:125]: done

File: qsv/test/org/qcmg/qsv/util/TestUtil.java (hunk 8 above)
Comment (ChristinaXu2017) [reply qsv/test/org/qcmg/qsv/util/TestUtil.java:851]: just follow the old method to create a file. But you are right, it is not necessary. I will delete this line.

File: qbasepileup/test/org/qcmg/qbasepileup/indel/IndelPileupForDeletionTest.java (hunk 9 above)
Comment (ChristinaXu2017) [reply qbasepileup/test/org/qcmg/qbasepileup/indel/IndelPileupForDeletionTest.java:161]: done

File: qpileup/test/org/qcmg/pileup/OptionsTest.java (hunk 10 above)
Comment (ChristinaXu2017) [reply qpileup/test/org/qcmg/pileup/OptionsTest.java:66]: done

File: qpileup/test/org/qcmg/pileup/OptionsTest.java (hunk 11 above)
Comment (ChristinaXu2017) [reply qpileup/test/org/qcmg/pileup/OptionsTest.java:83]: done

File: qpileup/test/org/qcmg/pileup/OptionsTest.java (hunk 12 above)
Comment (ChristinaXu2017) [reply qpileup/test/org/qcmg/pileup/OptionsTest.java:100]: done

**Pull request merged**